src/main/java/com/mongodb/hibernate/internal/translate/mongoast/command/aggregate/AstProjectStage.java

```diff
@@ -19,7 +19,14 @@
 import java.util.List;
 import org.bson.BsonWriter;
 
+import static com.mongodb.hibernate.internal.MongoAssertions.assertFalse;
+
 public record AstProjectStage(List<? extends AstProjectStageSpecification> specifications) implements AstStage {
+
+    public AstProjectStage {
+        assertFalse("Projection stage must have at least one specification", specifications().isEmpty());
+    }
+
```

Comment:
the `compact constructor` of Java record seems to be the place to insert our assertion logic. See https://mikemybytes.com/2022/02/16/java-records-and-compact-constructors/